src/HotChocolate/Data/src/Data/Projections/Extensions/ProjectionObjectFieldDescriptorExtensions.cs

```diff
@@ -205,10 +206,27 @@ private static FieldMiddleware CreateMiddleware<TEntity>(IProjectionConvention c
                 context = new MiddlewareContextProxy(context, selection, objectType);
             }
 
+            if (context.Operation.Definition.Operation == OperationType.Mutation
+                && context.Selection.Field.Type.NullableType() is ObjectType mutationPayloadType
```

Comment:
@michaelstaib  would this be clearer?
```suggestion
                && context.Selection.Field.Type.NamedType() is ObjectType mutationPayloadType
```